src/base/ShapesFactory.js

```diff
@@ -0,0 +1,33 @@
+/**
+ *
+ *  Copyright 2017 Adobe, Inc.
+ *
+ *     Licensed under the Apache License, Version 2.0 (the "License");
+ *     you may not use this file except in compliance with the License.
+ *     You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *     Unless required by applicable law or agreed to in writing, software
+ *     distributed under the License is distributed on an "AS IS" BASIS,
+ *     WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *     See the License for the specific language governing permissions and
+ *     limitations under the License.
+ *
+ */
+const ShapesFactory = {};
+ShapesFactory.shapes = [];
+ShapesFactory.registerShape = function (shapeName, shapeClass) {
+  if (ShapesFactory.shapes[shapeName] === undefined) {
+    ShapesFactory.shapes[shapeName] = shapeClass;
+  }
+}
+
+ShapesFactory.getShape = function (shapeName) {
+  if (ShapesFactory.shapes[shapeName] !== null && ShapesFactory.shapes[shapeName] !== undefined) {
```

Comment:
Assuming a shape with a value of `0` wouldn't be valid, so this could just be `if (ShapesFactory.shapes[shapeName])`

Reply:
+1, will do the change